### ISSUE TITLE
NormalizeWhitespace does not automatically insert a newline after documentation

### ISSUE THREAD

**Issue by JurjenBiewenga** [dotnet/roslyn#50626]
**Version Used**: Mircrosoft.CodeAnalysis.Common : 4.0.30319

**Steps to Reproduce**:

1. Emit a method using the following declaration: 
```cs
MethodDeclaration(
	PredefinedType(
		Token(SyntaxKind.IntKeyword)),
	Identifier("Method"))
.WithAttributeLists(
	SingletonList(
		AttributeList(
				SingletonSeparatedList<AttributeSyntax>(
					Attribute(
						IdentifierName("Pure"))))
			.WithOpenBracketToken(
				Token(
					TriviaList(
						Trivia(
							DocumentationCommentTrivia(
								SyntaxKind.SingleLineDocumentationCommentTrivia,
								SingletonList<XmlNodeSyntax>(
									XmlText()
										.WithTextTokens(
											TokenList(
												new []{
													XmlTextLiteral(
														TriviaList(
															DocumentationCommentExterior("///")),
														" Some cool documentation",
														" Some cool documentation",
														TriviaList())})))))),
					SyntaxKind.OpenBracketToken,
					TriviaList()))))
.WithModifiers(
	TokenList(
		Token(SyntaxKind.PublicKeyword)))
.WithBody(
	Block(
		SingletonList<StatementSyntax>(
			ReturnStatement(
				LiteralExpression(
					SyntaxKind.NumericLiteralExpression,
					Literal(1))))))
.NormalizeWhitespace());
```

**Expected Behavior**: 
Should automatically output a new line after the documentation
```cs
/// Some cool documentation
[Pure]
public int Method()
{
    return 1;
}
```

**Actual Behavior**: Does not output a new line after the documentation
```cs
/// Some cool documentation[Pure]
public int Method()
{
    return 1;
}
```

Manually adding a XmlTextNewLine token at the end of the documentation resolves the issue, not sure if this is the intended behavior.

**Comment by CyrusNajmabadi** [comment]
This does feel like a potential bug.  I would expect NormalizeWhitespace to insert this NewLine token if necessary.  I think we'd take a PR here if you wanted to contribute one :)

**Comment by Youssef1313** [comment]
I'm unable to reproduce with the following test:

```
        [Fact]
        [WorkItem(50626, "https://github.com/dotnet/roslyn/issues/50626")]
        public void TestNormalizeDocComment()
        {
            var code =
@"using System.Diagnostics.Contracts;

class C
{
  /// Doc comment
  [Pure]
  public void Method()
  {
  }
}";
            TestNormalizeDeclaration(code, code);
        }
```

Should it be written in some other way?

**Comment by Youssef1313** [comment]
@jaredpar I'm feeling this is not a bug. Would you be able to confirm and close if I'm correct?

**Comment by jaredpar** [comment]
@333fred probably better to say if this is a bug or not

**Comment by 333fred** [comment]
Our APIs are not hardened in the face of syntax trees we don't expect. @CyrusNajmabadi for a second opinion, but I don't think this a bug.

**Comment by CyrusNajmabadi** [comment]
Yeah, it's currently unclear to me if the tree that is being normalized is equivalent to a tree the parser would produce or not.  If it is equivalent, then this is a bug on our side.  If it not equivalent, then it's not a bug.

Note: equivalence here obviously doesn't refer to whitespace trivia (as that is what is being normalized).  However, i would expect that all *other* nodes/tokens need to be in the constructed tree that is being normalized.  Again, if all the nodes/tokens are there, it's a bug on our side to not normalize to something acceptable.  If they're nto all there, ti's an issue for the user.
--

That said, bug or not, the prioritization is up to you guys.  @Youssef1313 seems to have a very trivial (hyuck hyuck) workaround that can be used here to make forward progress on the issue.

**Comment by sharwell** [comment]
Normalize whitespace is only expected to have consistent behavior if the input syntax tree has the same form the compiler would produce when parsing the text of that tree. If a different tree is provided, the normalization can vary in unspecified ways. Therefore, I would consider this not a bug.
